qiskit/algorithms/gradients/base_estimator_gradient.py

```diff
@@ -45,11 +54,19 @@ def __init__(
                 The order of priority is: options in ``run`` method > gradient's
                 default options > primitive's default setting.
                 Higher priority setting overrides lower priority setting
+            supported_gates: The list of supported gates to decompose circuits. Defaults to None.
+            add_virtual_var: Whether to add virtual variables to circuits. This is used for the
+                parameter shift gradient. Defaults to False.
+            skip_pre_postprocess: Whether to skip the preprocess and postprocess. Defaults to False.
```

Comment:
This will need some more information before merging, like explaining that this can be set to true if the circuit has only plain parameters and is in the supported basis. But this can also wait until the end 🙂